src/vehicle.cpp

```diff
@@ -1084,29 +1007,29 @@ void vehicle::use_controls( const tripoint &pos, const bool remote_action )
         menu.addentry( manual_aim, true, 'w', _("Aim turrets manually") );
     }
     // toggle cameras
-    if( has_electronic_controls && (camera_on || ( has_camera && has_camera_control )) ) {
+    if( has_electronic_controls && (camera_on || ( has_part( "CAMERA" ) && has_part( "CAMERA_CONTROL" ) ) ) ) {
         menu.addentry( toggle_camera, true, 'M', camera_on ?
                        _("Turn off camera system") : _("Turn on camera system") );
     }
-    if( has_electronic_controls && has_plow ){
-        menu.addentry( toggle_plow, true, MENU_AUTOASSIGN, plow_on ?
-                       _("Turn plow off") : _("Turn plow on") );
-    }
-    if( has_electronic_controls && has_planter ){
-        menu.addentry( toggle_planter, true, 'P', planter_on ?
-                       _("Turn planter off") : _("Turn planter on") );
-    }
-    if( has_electronic_controls && has_scoop ) {
-        menu.addentry( toggle_scoop, true, 'S', scoop_on ?
-                       _("Turn off scoop system") : _("Turn on scoop system") );
-    }
-    if( has_electronic_controls && has_reaper ){
-        menu.addentry( toggle_reaper, true, 'H', reaper_on ?  _("Turn off reaper") : _("Turn on reaper") );
-    }
+
     menu.addentry( control_cancel, true, ' ', _("Do nothing") );
 
     menu.query();
 
+    auto toggle = [&]( const std::string &flag ) {
```

Comment:
This one won't work well for many common configurations:
* "Toggle lights" with active headlights and inactive floodlight
* Turrets in similar configuration
* Install recharger, toggle its state, install another, toggle, what is turned on and what off now?